server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1590,7 +1597,13 @@ public void setDefaultSearchPipeline(String defaultSearchPipeline) {
         if (FeatureFlags.isEnabled(SEARCH_PIPELINE)) {
             this.defaultSearchPipeline = defaultSearchPipeline;
         } else {
-            throw new SettingsException("Unsupported setting: " + DEFAULT_SEARCH_PIPELINE.getKey());
+            throw new SettingsException(
+                "Unable to update setting: "
+                    + DEFAULT_SEARCH_PIPELINE.getKey()
+                    + ". This is an experimental feature that is currently disabled, please enable the "
+                    + SEARCH_PIPELINE
+                    + " feature flag first."
+            );
```

Comment:
I'm leaning towards reverting this but don't have a strong opinion

Reply:
I will leave it since you have already made the change and it is improving the error message.

Reply:
why revering?

Reply:
Just because it's kind of out of scope of this PR. I won't revert though since I've already made the change

Reply:
Sorry I don't follow ...

Reply:
Basically @sohami pointed out that there's an existing mechanism to handle settings that are supposed to be gated by a feature flag, which I have changed the concurrent search settings to use. Here it looks like this `DEFAULT_SEARCH_PIPELINE` setting is not following that convention and the focus of my PR is not this setting so I was considering reverting the update to the exception message, but since I already improved the exception message I think it's good to keep it in regardless.

Reply:
Oh got it, thank you, understood